Refine GSI PM DA for the RRFS_SD model

hongli-wang opened the pull request:
## DESCRIPTION OF CHANGES: 
Modified below files for RRFS_SD PM DA
	modified:   anavinfo.rrfs_sd
	new file:   berror.rrfs_sd
	modified:   ../../scripts/exregional_run_analysis.sh
	modified:   ../../ush/config_defaults.sh
	modified:   ../../ush/templates/FV3LAM_wflow.xml

## TESTS CONDUCTED: 
This update is only affects GSI PM2.5 DA. In other worlds, it will not affect any meteorological DA capability. 
By default, NO PM DA (DO_SDDACYCLE="FALSE"), so there will be not PM GSI DA task is added into generated workflow *xml file. This is tested. 
When DO_SDDACYCLE="TRUE", a PM GSI task is added into the generated workflow *xml file. This is test. 

## DEPENDENCIES:
No. 

## DOCUMENTATION:
This PR updates the exist PM DA task in workflow.

hongli-wang committed:
Refine GSI PM DA for RRFS_SD model
	modified:   anavinfo.rrfs_sd
	new file:   berror.rrfs_sd
	modified:   ../../scripts/exregional_run_analysis.sh
	modified:   ../../ush/config_defaults.sh
	modified:   ../../ush/templates/FV3LAM_wflow.xml

hu5970 commented on ush/templates/FV3LAM_wflow.xml:
```diff
@@ -2251,6 +2331,9 @@ MODULES_RUN_TASK_FP script.
          <taskdep task="&PREP_CYC_TN;_prod{{ uscore_ensmem_name }}"/>
+         {%- if do_sddacycle %}
+         <taskdep task="&ANAL_SD_GSI_TN;_prod"/>
+         {%- endif %}
```
PM analysis has to run at the same time as conventional DA. This dependence should not be here.

hu5970 commented on ush/templates/FV3LAM_wflow.xml:
```diff
@@ -2212,6 +2212,86 @@ MODULES_RUN_TASK_FP script.
 {%- endif %}
 
 {%- if do_dacycle %}
+{%- if do_sddacycle %}
```
There is another option "DO_PM_DA". Please merge "do_sddacycle" and "DO_PM_DA" option as one option.

hongli-wang replied:
Thanks. I will look into this tomorrow.

hongli-wang replied:
@hu5970 
Please review the changes according to the discussion yesterday. 
Thanks,
Hongli

hongli-wang committed:
Modify FV3LAM_wflow.xml to remove dependence on SD_DA

hongli-wang committed:
Merge remote-tracking branch 'upstream/feature/RRFS_dev1' into dev1_pm_da

hongli-wang committed:
Update dependences on SD_DA

hongli-wang committed:
1. SD DA updates local tracer files
2. Update smoke,dust tracers
 Changes to be committed:
	modified:   scripts/exregional_run_analysis.sh
	modified:   scripts/exregional_run_postanal.sh
	modified:   ush/templates/FV3LAM_wflow.xml

hu5970 commented on ush/templates/FV3LAM_wflow.xml:
```diff
@@ -1346,29 +1346,8 @@ MODULES_RUN_TASK_FP script.
-         {%- endif %}
-         {%- endif %}
+         <timedep><cyclestr offset="&START_TIME_PROCPM;">@Y@m@d@H@M00</cyclestr></timedep>
+         <taskdep task="&PROCESS_PM_TN;_spinup"/>
```
should have dependence to 
`<taskdep task="&PREP_CYC_TN;_#type#{{ uscore_ensmem_name }}"/>`

hu5970 commented on ush/templates/FV3LAM_wflow.xml:
```diff
@@ -2257,6 +2236,48 @@ MODULES_RUN_TASK_FP script.
+    <dependency>
+       <and>
+         <timedep><cyclestr offset="&START_TIME_PROCPM;">@Y@m@d@H@M00</cyclestr></timedep>
+         <taskdep task="&PROCESS_PM_TN;_prod"/>
```
Need to add dependence to prep_cyc task.

hu5970 commented:
Also, "DO_SDDACYCLE" will not be used. Could you remove it?

hongli-wang committed:
Remove DO_SDDACYCE
Change dependece to PREP_CYC_TN

guoqing-noaa commented on fix/gsi/anavinfo.rrfs_sd:
```diff
@@ -55,11 +56,13 @@ control_vector::
 !var     level  itracer crtm_use   type    orig_name
  smoke    65      1       12     smoke     smoke
  dust     65      1       12     dust      dust
+ coarsepm 65      1       12     coarsepm  coarsepm
```
This is what confused me: coarseepm is added here, but in your GSI PR, you removed src/gsi/setuppm2_5.f90 (line 281). Will corseepm be used or not? Or is this just a placeholder for future?

hongli-wang replied:
> This is what confused me: coarseepm is added here, but in your GSI PR, you removed src/gsi/setuppm2_5.f90 (line 281). Will corseepm be used or not? Or is this just a placeholder for future?

@guoqing-noaa 
1. An obs operator does not have to use all the variables in control/state
vector
2. Pm10 AOD assimilation will  need it.
3. EnVar is under consideration so the PM da can be combined with dbz DA
with RRFSE ens.

hongli-wang commented:
1. An obs operator does not have to use all the variables in control/state
vector
2. Pm10 AOD assimilation will  need it.
3. EnVar is under consideration so the PM da can be combined with dbz DA
with RRFSE ens.

On Tue, Sep 26, 2023 at 4:20 PM Guoqing Ge ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In fix/gsi/anavinfo.rrfs_sd
> <https://github.com/NOAA-GSL/regional_workflow/pull/569#discussion_r1337838619>
> :
>
> >  ::
>
>  chem_guess::
>  !var     level  itracer crtm_use   type    orig_name
>   smoke    65      1       12     smoke     smoke
>   dust     65      1       12     dust      dust
> + coarsepm 65      1       12     coarsepm  coarsepm
>
> This is what confused me: coarseepm is added here, but in your GSI PR, you
> removed src/gsi/setuppm2_5.f90 (line 281). Will corseepm be used or not? Or
> is this just a placeholder for future?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-GSL/regional_workflow/pull/569#pullrequestreview-1645289122>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMXB44WS34TE23ILVHC2NK3X4NIJNANCNFSM6AAAAAA5GSCLGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

hongli-wang commented:
> Also, "DO_SDDACYCLE" will not be used. Could you remove it?

@hu5970 
Had modified the code. Please review again. Thanks.
